lib/index.js

```diff
@@ -11,21 +25,36 @@ const DEFAULTS = {
   appSecret: '',
   agentVersion: 'unknown',
   endpoint: 'https://enginex.kadira.io',
-  hostname: 'localhost',
+  hostname: hostname(),
   clockSyncInterval: 1000 * 60,
   dataFlushInterval: 1000 * 10,
+  retryOptions: {
+    maxRetries: 3, // Same as the previous 4 not counting the first try.
+  }
 };
 
 // exporting this for if we need to get this as a NPM module.
-export class Kadira {
+export class Kadira extends EventEmitter2 {
+  _id = uuid();
+  _supportedFeatures = SupportedFeatures;
+  _allowedFeatures = {};
+  _jobQueue = [];
```

Comment:
The idea is to queue tasks so the agent has time to complete them if multiple of them are sent at the same time, it would go one by one. I guess it could also be used for race conditions too.

The agent's code would be responsible for handling those right now. Will try improving the interface on that so it is less error prone.

Reply:
I added a `jobDequeue` method so it is more clear.

Reply:
I had some changes made on the agent for websockets, but I lost them, will have to redo.

Reply:
Removed the job queue as we will handle it on the agent.